frontend/ui/data/keyboardlayouts/fr_keyboard.lua

```diff
@@ -1,70 +1,50 @@
-            { "M",        "m",    "’",    "↓",    "Ь",    "ь",    "Ю",    "ю",    "Ç",    "ç",    "…",    "¨", },
-            { label = "Backspace",
-              icon = "resources/icons/appbar.clear.reflect.horizontal.png",
-              width = 1.5
-            },
-        },
-        -- fourth row
-        {
-            { "Sym",     "Sym",  "ABC",  "ABC",  "Sym",  "Sym",  "ABC",  "ABC",  "Sym",  "Sym",  "ABC",  "ABC",
-              width = 1.5},
-            { label = "IM",
-              icon = "resources/icons/appbar.globe.wire.png",
-            },
-            { "Äéß",     "Äéß",  "Äéß",  "Äéß",  "Äéß",  "Äéß",  "Äéß",  "Äéß",  "Äéß",  "Äéß",  "Äéß",  "Äéß", },
-            { label = "space",
-              " ",        " ",    " ",    " ",    " ",    " ",    " ",    " ",    " ",    " ",    " ",    " ",
-              width = 3.0},
-            { ",",        ",",    "“",    "←",    ",",    ",",    "Ё",    "ё",    "Ũ",   "ũ",    ",",    ",", },
-            { ".",        ".",    "”",    "→",    ".",    ".",    ".",    ".",    "Ĩ",   "ĩ",    ".",    ".", },
-            { label = "Enter",
-              "\n",       "\n",   "\n",   "\n",   "\n",   "\n",   "\n",   "\n",   "\n",   "\n",   "\n",   "\n",
-              icon = "resources/icons/appbar.arrow.enter.png",
-              width = 1.5,
-            },
-        },
-    },
+-- Start with the english keyboard layout
+local fr_keyboard = dofile("frontend/ui/data/keyboardlayouts/en_keyboard.lua")
+
+-- Swap the four AZWQ keys (only in the lowercase and
+-- uppercase letters layouts) to change it from QWERTY to AZERTY
+local keys = fr_keyboard.keys
+keys[1][1][1], keys[2][1][1] = keys[2][1][1], keys[1][1][1] -- Q <> A
+keys[1][1][2], keys[2][1][2] = keys[2][1][2], keys[1][1][2] -- q <> a
+keys[1][2][1], keys[3][2][1] = keys[3][2][1], keys[1][2][1] -- W <> Z
+keys[1][2][2], keys[3][2][2] = keys[3][2][2], keys[1][2][2] -- w <> z
+
+-- Insert an additional key at the end of 2nd row for M
+table.insert(keys[2],
+           --  1           2       3       4       5       6       7       8       9       10      11      12
+            { "M",        "m",    "§",    "+",    "Д",    "д",    "Э",    "э",    "Œ",    "œ",    "Ő",    "ő", }
+)
+-- And swap the english M on the 3rd row to ','
+keys[3][8][1] = ","
+keys[3][8][2] = ","
+-- And swap the ',' on the 4th row (an extended key including a popup) to ';'
+local en_com = keys[4][5][1]
+en_com[1] = ";"
+en_com.north = "," -- and swap the ';' there to ','
+
+-- Swap ê and ë (and the like) in the keyboard popups, so the
+-- common french accentuated chars are all on the upper row.
+local popups = {
+    keys[1][1][1], -- A
+    keys[1][1][2], -- a
+    keys[1][3][1], -- E
+    keys[1][3][2], -- e
+    keys[1][7][1], -- U
+    keys[1][7][2], -- u
+    keys[1][8][1], -- I
+    keys[1][8][2], -- i
+    keys[1][9][1], -- O
+    keys[1][9][2], -- o
 }
+for _, popup in ipairs(popups) do
+    if popup.east and popup.north then -- not all implemented yet
```

Comment:
We can just add a couple of stubs instead? It's the other options that require thought but the basic òöóô-type ones should be fine.

Reply:
PS I'll do that now. Shouldn't take more than a few mins.

Reply:
Dunno (yet) what you mean by stub, but I'll see. But may be no need to add anything.

Anyway, I planned to remove that once you have implemented them for all.
So, no real need for anything more.

(And anyway, it's not even needed (and a bit wrong with my intent) : `local toto = "zzz" ; toto.north` returns nil, when I thought it would crash with "attempt to index a string" - but no - and if it's not a table it's a no-op)

Reply:
Alright, so let's take out the if condition and merge it. ;-)

Reply:
Alright. (so, UIO are done :)

Reply:
Just a stub; it'd require some more research to call it done. :-)